browser/extensions/brave_extension_management.cc

```diff
@@ -43,11 +54,28 @@ BraveExtensionManagement::~BraveExtensionManagement() {
 
 void BraveExtensionManagement::RegisterBraveExtensions() {
 #if BUILDFLAG(ENABLE_TOR)
+  const bool isTorEnabled = !tor::TorProfileService::IsTorDisabled();
   const base::CommandLine& command_line =
       *base::CommandLine::ForCurrentProcess();
-  if (!command_line.HasSwitch(switches::kDisableTorClientUpdaterExtension) &&
-      !profile_->AsTestingProfile())
+  if (isTorEnabled &&
+      !command_line.HasSwitch(switches::kDisableTorClientUpdaterExtension) &&
+      !profile_->AsTestingProfile()) {
     g_brave_browser_process->tor_client_updater()->Register();
+  }
+#endif
+}
+
+void BraveExtensionManagement::CleanupBraveExtensions() {
+#if BUILDFLAG(ENABLE_TOR)
+  // Delete tor binaries if tor is disabled by gpo.
+  if (tor::TorProfileService::IsTorDisabled()) {
+    ProfileManager* profile_manager = g_browser_process->profile_manager();
+    base::FilePath tor_component_dir =
+        profile_manager->user_data_dir().AppendASCII(
+            extensions::kTorClientComponentId);
+    g_brave_browser_process->tor_client_updater()->GetTaskRunner()->PostTask(
```

Comment:
The delete should be encapsulated in the tor_client_updater - `tor_client_updater()->Cleanup()`

Reply:
Refactored. Done.